Source/DafnyCore/Generic/ErrorRegistry.cs

```diff
@@ -26,6 +26,7 @@ public record DafnyAction(string Title, IReadOnlyList<DafnyCodeActionEdit> Edits
 
 public static class ErrorRegistry {
 
+  // Replace any NoneId by ParseErrors.p_... or ResolutionErrors.r_
```

Comment:
I came across warnings and errors which were using ErrorRegistry.NoneId as identifier. However, at first, I did not know how to put an identifier that has some meaning, so I had to do research and found out that meaningful identifiers are actually defined in ParseErrors.p_.... so I created the same for ResolutionErrors.r_.... and put a comment so that if a resolution error is using ErrorRegistry.NoneId and someone wants to create a rich error (e.g. that contains extended error text or a code action), they know where to go and what to do.

Reply:
Got it.